R/profile_wrapper.R

```diff
@@ -141,26 +152,72 @@ profile_wrapper <- function(mydir, model_settings) {
     vec <- c(low, high)
     num <- sort(vec, index.return = TRUE)$ix
 
-    profile <- r4ss::profile(
-      dir = profile_dir,
-      oldctlfile = "control.ss_new",
-      newctlfile = model_settings$newctlfile,
-      linenum = model_settings$linenum,
-      string = para,
-      profilevec = vec,
-      usepar = model_settings$usepar,
-      globalpar = model_settings$globalpar,
-      parlinenum = model_settings$parlinenum,
-      parstring = model_settings$parstring,
-      saveoutput = model_settings$saveoutput,
-      overwrite = model_settings$overwrite,
-      whichruns = model_settings$whichruns,
-      prior_check = model_settings$prior_check,
-      read_like = model_settings$read_like,
-      exe = model_settings$exe,
-      verbose = model_settings$verbose,
-      extras = model_settings$extras
-    )
+    # backup original control.ss_new file for use in second half of profile
+    file.copy(file.path(profile_dir, model_settings$oldctlfile), 
+      file.path(profile_dir, "backup_oldctlfile.ss"),
+      overwrite = model_settings$overwrite)
+    # backup original par file for use in second half of profile
+    # if usepar = TRUE
+    file.copy(file.path(profile_dir, "ss.par"), 
+      file.path(profile_dir, "backup_ss.par"),
+      overwrite = model_settings$overwrite)
+
+    # loop over down, then up
```

Comment:
At first I was very anti loop when I saw this code and I wanted to find a way to not have a loop. Then looking over it, the code is not complicated and it is easy to read. The part I am still struggling with is that there are duplicated lines. I find that bugs are easily introduced when function calls are repeated. I prefer `ifelse` statements to make it clear that all iterations of the loop use the same function call but with slightly different arguments. For example, ⚠️ @chantelwetzel-noaa brace yourself for a nested function call,
```
    for (iprofile in 1:2) {
      whichruns <- intersect(
        x = if (!is.null(model_settings$whichruns)) {
          model_settings$whichruns
        } else {
          seq(1, length(c(low, high)))
        },
        y = which(vec %in% if (iprofile == 1){low} else{high})
      )
```
We don't have to do something this terrible to read but at a minimum we could use
```
    if (model_settings$verbose) {
      message("Running profiles ", ifelse(iprofile == 1, "down", "up"), " from estimated value:",
        paste(whichruns, collapse = " "), "out of ", length(vec))
    }
```
rather than two calls to `message()`.

Reply:
What do you think about eliminating the messages here entirely?  Across the functions used in this package, a bunch of items are printed on the screen by r4ss functions. I could probably find ways to sink it somewhere else but it feels like too much trouble right now.  I think these kind of simple running messages get lost and don't end up being helpful to people.

Reply:
I added these messages while debugging without remembering the `r4ss::profile()` provides something similar. I should have circled back to remove them. I endorse deleting.

Reply:
I also agree with deleting them. @chantelwetzel-noaa I also think that most of this stuff gets ignored and we should focus messages for the user when they need to do stuff rather than what step of the code the process is currently on.

Reply:
Good point @kellijohnson-NOAA. I think the challenge is that this package provides a lot of layers of separation from the 
`run_diagnostics()` calls `profile_wrapper()` which calls `r4ss::profile()` which calls `r4ss::run()`, so when things go wrong as they did for me, it can be hard to figure out what's going on.

I think that's the point of having a `verbose` setting. If things are working, you should leave `verbose = FALSE` and enjoy getting all the wonderful results that this package provides. When things don't work, it's a lot nicer to turn on `verbose` than to have to add a bunch of browser() statements.

Reply:
Post-document deadline I would love to touch base to understand better the challenges you had with petrale sole and what additional revisions could be done to make this easier in the future.

Reply:
@iantaylor-NOAA I have never thought of verbose as a tool for debugging. I always thought that it was messages for the user not messages for the developers. I think this is a design question that we should address in a different issue, maybe on PEPtools and then apply the outcome across all packages.

Reply:
Let's indeed circle back to this later. I hope that I can remember the issues that I had.

I was looking for more information as a user. I think there are times that your inputs are bad and with better message you can fix it rather than just email the package developers every time something doesn't work right away. For sure the message in question here is not valuable, but I think other places would be (e.g. #28).